tests/media_filename_test.go

```diff
@@ -0,0 +1,152 @@
+package tests
+
+import (
+	"fmt"
+	"mime"
+	"testing"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/client"
+	"github.com/matrix-org/complement/internal/data"
+)
+
+const asciiFileName = "ascii"
+const unicodeFileName = "\xf0\x9f\x90\x94"
+
+func TestMediaFilenames(t *testing.T) {
+	deployment := Deploy(t, b.BlueprintFederationOneToOneRoom)
+	defer deployment.Destroy(t)
+
+	alice := deployment.Client(t, "hs1", "@alice:hs1")
+	bob := deployment.Client(t, "hs2", "@bob:hs2")
+
+	t.Run("Parallel", func(t *testing.T) {
+
+		t.Run("ASCII", func(t *testing.T) {
+			t.Parallel()
+
+			// sytest: Can upload with ASCII file name
+			t.Run("Can upload with ASCII file name", func(t *testing.T) {
+				t.Parallel()
+
+				alice.UploadContent(t, data.TestPngImage, asciiFileName, "image/png")
+			})
+
+			// sytest: Can download file '$filename'
+			for _, filename := range []string{"ascii", "name with spaces", "name;with;semicolons"} {
+
+				// else linter will scream
+				var filename = filename
```

Comment:
This isn't a linter thing. Without this, the test function will be called 3 times with the last string in the array `"name;with;semicolons"` as you're just holding a reference to the variable which is overwritten on each loop.

Reply:
I know, the comment was partially in humor, but i'll edit it to explain why this line is here.